.github/workflows/publish_website.yml

```diff
@@ -0,0 +1,31 @@
+# Builds the website sourced from docs/website/ using `mkdocs` and pushes
+# to the gh-pages branch for publishing on GitHub Pages.
+#
+# See https://squidfunk.github.io/mkdocs-material/publishing-your-site/
+
+name: Publish Website
+
+on:
+  workflow_dispatch:
+  push:
+    branches:
+      - main
```

Comment:
Not sure if we want to do something different here (schedule, only on changes to the `docs/website/` folder, etc.). WDYT?

Reply:
If we know it's only taking source from docs/website, let's add that to the path exclusions, yeah

Reply:
I haven't tested how this action behaves when no changes are made to the generated site. Presumably it just doesn't commit anything, but that would be good to verify.

The source should be exclusively `docs/website/` for now. If we add some external build (like the dialect reference pages or IR conversion examples) then that will no longer be true.

Reply:
Yeah that is a little odd because git usually doesn't allow empty commits, but it looks like the binary sitemap file changes. If we limit this to running on updates to docs/website then we shouldn't get that issue often, so I think it's fine. We can revisit when we expand things. Probably could have it commit to the local gh-pages branch and check that anything actually interesting changed there before deploying

Reply:
Ok cool, path filtering on the `push` trigger works:
![image](https://user-images.githubusercontent.com/4010439/117487042-621f5680-af1f-11eb-9ad5-7c1b54c3f02e.png)

(unrelated change did not trigger the action)

https://github.com/ScottTodd/iree/commits/docs-publish-test